x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -1012,4 +1016,68 @@ static DeprecationIssue checkTracingApmSettings(
             DeprecationIssue.Level.CRITICAL
         );
     }
+
+    static DeprecationIssue checkSourceModeInMappingsAndTemplates(
+        final Settings settings,
+        final PluginsAndModules pluginsAndModules,
+        final ClusterState clusterState,
+        final XPackLicenseState licenseState
+    ) {
+        List<String> indexes = new ArrayList<>();
+        for (String indexName : clusterState.metadata().getIndices().keySet()) {
+            IndexMetadata indexMetadata = clusterState.metadata().getIndices().get(indexName);
+            if (SourceFieldMapper.onOrAfterDeprecateModeVersion(indexMetadata.getCreationVersion())) {
```

Comment:
I don't think we need to emit warnings for _source.mode mapping attribute usage in indices, because these are not actionable. The source mode can't be changed after the index has been created.  This is different for templates.

Reply:
Note that create index api will return emit a deprecation warning if _source.mode mapping attribute is used. This is visible as response header and in the deprecation log file.